### PR TITLE
[MRG] Use pseudo-inverse in Covariance

### LINKED ISSUE #204: Update covariance
In the `Covariance` algorithm, we currently do an inverse of the covariance matrix, and the latter could be singular. We need to decide whether to return an error or return the pseudo inverse in this case.

### PR DESCRIPTION
**Pull request by wdevazelhes**
Fixes #204 

This PR uses the pseudo inverse in Covariance, rather than the inverse. See https://github.com/metric-learn/metric-learn/issues/204#issuecomment-492336040

### CODE REVIEW

File: metric_learn/covariance.py [hunk 1]
```diff
@@ -35,11 +35,11 @@ def fit(self, X, y=None):
     y : unused
     """
     X = self._prepare_inputs(X, ensure_min_samples=2)
-    M = np.cov(X, rowvar = False)
-    if M.ndim == 0:
+    M = np.atleast2d(np.cov(X, rowvar = False))
```
Comment (wdevazelhes) [inline metric_learn/covariance.py]: I added an atleast2d since we don't want the metric to be scalar

File: metric_learn/covariance.py [hunk 2]
```diff
@@ -35,11 +35,11 @@ def fit(self, X, y=None):
...
+    if len(M) == 1:
```
Comment (wdevazelhes) [inline metric_learn/covariance.py]: Because of the atleast2d, this needs to change

File: metric_learn/covariance.py (hunk 1 above)
Comment (perimosocordiae) [reply metric_learn/covariance.py]: While we're here, remove the spaces in the `rowvar` keyword argument.

File: metric_learn/covariance.py (hunk 2 above)
Comment (perimosocordiae) [reply metric_learn/covariance.py]: Can we be more explicit and check `M.size == 1`? Would it be possible for M to be a column vector here?

File: metric_learn/covariance.py (hunk 1 above)
Comment (wdevazelhes) [reply metric_learn/covariance.py]: I agree, done

File: metric_learn/covariance.py (hunk 2 above)
Comment (wdevazelhes) [reply metric_learn/covariance.py]: Ah yes you're right thanks
I thought that len counted the number of elements but indeed `len(np.array([[1, 2, 3]])) == 1`

Comment (wdevazelhes) [reply metric_learn/covariance.py]: done

**Comment by wdevazelhes** [comment]
I think it should be good to merge now

File: metric_learn/covariance.py [hunk 3]
```diff
@@ -14,6 +14,7 @@
 
 from .base_metric import MahalanobisMixin
 from ._util import transformer_from_metric
+import scipy
```
Comment (perimosocordiae) [inline metric_learn/covariance.py]: Style nitpick: move this import up to line 13 (after the numpy import but before the sklearn)

**Comment by bellet** [comment]
Maybe fix the small style issue so we can merge?

File: metric_learn/covariance.py (hunk 3 above)
Comment (wdevazelhes) [reply metric_learn/covariance.py]: That's right, done

**Comment by wdevazelhes** [comment]
> Maybe fix the small style issue so we can merge?

Yep, sorry, it's done now I think we can merge

**Pull request merged**